WikiWatershed/model-my-watershed
# Unit Conversion: Draw and Analyze

## Overview

Adds a core module for converting units. Uses that core module to display favored units for the Draw and Analyze stages. All the work has been done according to [this source document](https://docs.google.com/spreadsheets/d/1Hn_PmhFz1mpDg79ZsF_gP1lZzVNCRvWQ2THsDhG6KBs).

Connects #3013 
Connects #3015 

### Demo

![image](https://user-images.githubusercontent.com/1430060/48449476-98c1ef00-e770-11e8-8582-8544f7e67b81.png)

![image](https://user-images.githubusercontent.com/1430060/48449496-b2633680-e770-11e8-8112-ccb1367f7213.png)

![image](https://user-images.githubusercontent.com/1430060/48449688-6bc20c00-e771-11e8-8c79-dea4ba9a1318.png)

etc.

### Notes

The work for the Model stage is upcoming, so #3015 isn't completely satisfied. Also, we'll have to think about the implications for existing scenarios, which may use `m<sup>2</sup>` terminology instead of the newer `m²`.

## Testing Instructions

* Check out this branch and `bundle`
* Go to [:8000/](http://localhost:8000/)
* Set `US Customary` as your favored unit scheme
* Draw a very large shape, the size of a small state, so you get an error message saying it is too big. Ensure the units in the error message are US Customary.
* Draw a smaller shape. Wait for Analyze to complete.
* Ensure that all the units are shown in US Customary.
* Compare the same shape on staging. Ensure that the values translate reasonably.
* Go to your account and set unit scheme to Metric.
* Draw the same shape. Wait for Analyze to complete.
* Ensure that all the units are shown in Metric.
